### PR TITLE
Move background settings from dashboard app to Appearance and accessibility settings

### PR DESCRIPTION
**Pull request by GretaD**
For https://github.com/nextcloud/server/issues/33736

### Screenshot

image

### To Do
- [x] move the code into a component
- [x] Remove the old component from dashboard
- [x] Re-route from dashboard to theming
- [x] Make sure that initial state is present in the theming app
- [x] Adjust default theme background image to pull in the core config value https://github.com/nextcloud/server/pull/33568/files#diff-2581e3cecfba06bf2e50e8f4e018ad4730f0c9179953ea9fc3e74fb0c6e7e0e5R237-R249

### CODE REVIEW

File: apps/theming/src/UserThemes.vue [hunk 1]
```diff
@@ -23,16 +23,47 @@
 				type="font"
 				@change="changeFont" />
 		</div>
+		<div class="background-selector">
+			<button class="background filepicker"
+				:class="{ active: background === 'custom' }"
+				tabindex="0"
+				@click="pickFile">
+				{{ t('dashboard', 'Pick from Files') }}
```
Comment (juliushaertl) [inline apps/theming/src/UserThemes.vue:31]:
```suggestion
				{{ t('theming', 'Pick from Files') }}
```

Comment (juliushaertl) [reply apps/theming/src/UserThemes.vue:31]: Same for the others ;)

File: apps/theming/src/UserThemes.vue [hunk 2]
```diff
@@ -23,16 +23,47 @@
 				type="font"
 				@change="changeFont" />
 		</div>
+		<div class="background-selector">
```
Comment (juliushaertl) [inline apps/theming/src/UserThemes.vue:26]: Maybe we could  still keep this in a separate component?

File: apps/theming/src/UserThemes.vue [hunk 3]
```diff
@@ -157,6 +210,49 @@ export default {
...
+			image.src = this.backgroundImage
+		},
+		async setDefault() {
+			this.loading = 'default'
+			const result = await axios.post(generateUrl('/apps/theming/background/default'))
+			this.update(result.data)
+		},
+		async setShipped(shipped) {
+			this.loading = shipped
+			const result = await axios.post(generateUrl('/apps/theming/background/shipped'), { value: shipped })
```
Comment (juliushaertl) [inline apps/theming/src/UserThemes.vue:235]: - [ ] Backend: Move routes for setting the background to /apps/theming

File: apps/theming/src/UserThemes.vue [hunk 4]
```diff
@@ -23,16 +23,47 @@
...
+import { generateOcsUrl, generateUrl } from '@nextcloud/router'
 import { loadState } from '@nextcloud/initial-state'
-import axios from '@nextcloud/axios'
 import SettingsSection from '@nextcloud/vue/dist/Components/SettingsSection'
 
 import ItemPreview from './components/ItemPreview'
+import axios from '@nextcloud/axios'
+import getBackgroundUrl from './helpers/getBackgroundUrl'
+import prefixWithBaseUrl from './helpers/prefixWithBaseUrl'
+const shippedBackgroundList = loadState('theming', 'shippedBackgrounds')
```
Comment (juliushaertl) [inline apps/theming/src/UserThemes.vue:66]: - [ ] Backen: Make sure that initial state is present in the theming app

File: apps/theming/src/helpers/prefixWithBaseUrl.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { generateFilePath } from '@nextcloud/router'
+
+export default (url) => generateFilePath('theming', '', 'img/') + url
```
Comment (juliushaertl) [inline apps/theming/src/helpers/prefixWithBaseUrl.js:25]: We need to move the background images then from apps/dashboard/img/ to apps/theming/img ...

File: apps/theming/src/UserThemes.vue (hunk 2 above)
Comment (juliushaertl) [inline apps/theming/src/UserThemes.vue:26]: Could need some heading and short description about what the image is about.

File: apps/theming/src/UserThemes.vue (hunk 3 above)
Comment (GretaD) [reply apps/theming/src/UserThemes.vue]: i tried to do that myself, please check :shark:

File: apps/dashboard/appinfo/routes.php [hunk 6]
```diff
@@ -29,8 +29,8 @@
 		['name' => 'dashboard#index', 'url' => '/', 'verb' => 'GET'],
 		['name' => 'dashboard#updateLayout', 'url' => '/layout', 'verb' => 'POST'],
 		['name' => 'dashboard#updateStatuses', 'url' => '/statuses', 'verb' => 'POST'],
-		['name' => 'dashboard#getBackground', 'url' => '/background', 'verb' => 'GET'],
-		['name' => 'dashboard#setBackground', 'url' => '/background/{type}', 'verb' => 'POST'],
+		['name' => 'theming#getBackground', 'url' => '/background', 'verb' => 'GET'],
```
Comment (juliushaertl) [inline apps/dashboard/appinfo/routes.php:32]: We basically need to move the controller methods and this entries to apps/theming as well ;)

File: apps/theming/lib/Controller/UserThemeController.php [hunk 7]
```diff
@@ -124,4 +133,54 @@ private function validateTheme(string $themeId): ITheme {
...
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function getBackground(): Http\Response {
+		$file = $this->backgroundService->getBackground();
+		if ($file !== null) {
+			$response = new FileDisplayResponse($file, Http::STATUS_OK, ['Content-Type' => $file->getMimeType()]);
+			$response->cacheFor(24 * 60 * 60);
```
Comment (ChristophWurst) [inline apps/theming/lib/Controller/UserThemeController.php:145]: should we go for immutable? https://github.com/nextcloud/mail/pull/7048 @CarlSchwan :eyes:

**Review by ChristophWurst — APPROVED** [review]
:+1: :elephant:

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by szaimen** [comment]
Conflicts...

**Comment by PVince81** [comment]
ok, let me rebase and rebuild...

**Comment by PVince81** [comment]
@Pytal was this actually finished ? the label still says "2 to develop", please adjust accordingly and merge if applicable once ready

**Comment by szaimen** [comment]
> @Pytal was this actually finished ? the label still says "2 to develop", please adjust accordingly and merge if applicable once ready

I added the label. It was finished

**Comment by szaimen** [comment]
Failing tests seem related

**Comment by szaimen** [comment]
CI failure unrelated

**Pull request merged**